data/pigui/modules/new-game-window/game-param.lua

```diff
@@ -0,0 +1,40 @@
+-- Copyright © 2008-2023 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+local utils = require 'utils'
+local GameParam = utils.class("pigui.modules.new-game-window.game-param")
+
+setmetatable(GameParam, { __index = function (self, key) assert(false, tostring(self) .. ": unexpected key - " .. tostring(key)) end })
+
+function GameParam:Constructor(name, path)
+	assert(name)
+	assert(path)
+	self.name = name
+	self.path = path
+	-- some parameters can have arbitrary lock modes
+	-- for example, you block ship selection with several initial models
+	self.lockModes = { LOCKED = true, UNLOCKED = true }
+	self.lock = "UNLOCKED"
+	self.value = nil
+	self.hidden = false
+	self.noName = false
+end
+
+function GameParam:isValid()          assert(false, tostring(self.name) .. ":isValid() should be overridden.") end
+function GameParam:draw()             assert(false, tostring(self.name) .. ":draw() Should be overridden.") end
+function GameParam:fromSaveGame()     assert(false, tostring(self.name) .. ":fromSave() Should be overridden.") end
+function GameParam:fromStartVariant() assert(false, tostring(self.name) .. ":fromStartVariant() Should be overridden.") end
```

Comment:
These function 'prototypes' should have their expected arguments in the argument list for completion's sake.

Reply:
Done!